blogposts/2020/introducing-codeintel-db.md

```diff
@@ -0,0 +1,65 @@
+
+## Notes for site admins
+
+If you are an active user of LSIF, there are a few changes that may subtly affect the behavior of the code intelligence backend. Our enterprise users operate at incredibly different scales and is difficult to ensure that a change has zero impact on all existing customers.
+
+#### Connecting to an external database
+
+It has always been possible to target an [external database](https://docs.sourcegraph.com/admin/external_database) instance instead of using the Dockerized version that ships with the Sourcegraph instance. This is useful when you already have a method to provision, backup, and restore database instances (for example, using AWS's RDS or GCP's CloudSQL).
+
+The [3.20 -> 3.21 upgrade guides](https://docs.sourcegraph.com/admin/updates) call out instructions on how to do the same for the new code intelligence database. In the same way you configure your primary database instance with `PG*` environment variables, you configure the code intelligence database instance with `CODEINTEL_PG*` environment variables.
+
+**Note** that if you configure both databases externally, they **must not** point to the exact same database. The frontend service will refuse to start with such a configuration.
+
+#### Data retention policy
+
+The [3.21 changelog](https://github.com/sourcegraph/sourcegraph/blob/3.21/CHANGELOG.md) details the introduction of a new environment variable, `PRECISE_CODE_INTEL_MAX_DATA_AGE`, on the precise-code-intel-bundle-manager service.
+
+Previously, the bundle manager stored all LSIF data on disk as SQLite files. This allowed us to run custodial processes which removed old data, but only when disk pressure was high. Since we've migrated this data to a Postgres instance, there is a less clear-cut method to determine how much data we need to free in order to keep the desired amount of headroom.
+
+Disks do not grow on trees and we at Sourcegraph are well aware of that fact. In order to ensure we didn't accidentally slip an infinite disk space requirement to the Sourcegraph 3.21 release, we introduced an alternate data retention policy to replace the old full-disk trigger. Any data that is older than `PRECISE_CODE_INTEL_MAX_DATA_AGE` and is not used to serve code intelligence queries for the tip of a default branch will be automatically flushed from the code intelligence database periodically.
+
+We believe that this will give us roughly the same behavior as the previous versions, and can be tuned to match expected behavior at your instance's scale. The default value for this environment variable is `720h` (30 days), which is likely adequate for the majority of instances. This value should be adjusted if the size of your index uploads are large and your commit frequency is high.
```

Comment:
```suggestion
We believe that this will give us roughly the same behavior as the previous versions, and can be tuned to match expected behavior at the scale of your instance. The default value for this environment variable is `720h` (30 days), which is likely adequate for the majority of instances. This value should be adjusted if the size of your index uploads are large and your commit frequency is high.
```